crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -211,67 +188,6 @@ impl TryFrom<types::SetupMandateRequestData> for types::ConnectorCustomerData {
 
 #[allow(clippy::too_many_arguments)]
 impl types::SetupMandateRouterData {
-    pub async fn decide_flow<'a, 'b>(
-        &'b self,
-        state: &'a AppState,
-        connector: &api::ConnectorData,
-        maybe_customer: &Option<domain::Customer>,
-        confirm: Option<bool>,
-        call_connector_action: payments::CallConnectorAction,
-        merchant_account: &domain::MerchantAccount,
-        key_store: &domain::MerchantKeyStore,
-        profile_id: Option<String>,
-    ) -> RouterResult<Self> {
-        match confirm {
-            Some(true) => {
-                let connector_integration: services::BoxedConnectorIntegration<
-                    '_,
-                    api::SetupMandate,
-                    types::SetupMandateRequestData,
-                    types::PaymentsResponseData,
-                > = connector.connector.get_connector_integration();
-                let mut resp = services::execute_connector_processing_step(
-                    state,
-                    connector_integration,
-                    self,
-                    call_connector_action,
-                    None,
-                )
-                .await
-                .to_setup_mandate_failed_response()?;
-
-                let payment_method_type = self.request.payment_method_type;
-
-                let (pm_id, payment_method_status) = Box::pin(tokenization::save_payment_method(
-                    state,
-                    connector,
-                    resp.to_owned(),
-                    maybe_customer,
-                    merchant_account,
-                    payment_method_type,
-                    key_store,
-                    resp.request.amount,
-                    Some(resp.request.currency),
-                    profile_id,
-                ))
-                .await?;
-
-                resp.payment_method_id = pm_id.clone();
-                resp.payment_method_status = payment_method_status;
-
-                Ok(mandate::mandate_procedure(
-                    state,
-                    resp,
-                    maybe_customer,
-                    pm_id,
-                    connector.merchant_connector_id.clone(),
-                )
-                .await?)
-            }
-            _ => Ok(self.clone()),
-        }
-    }
-
     async fn update_mandate_flow(
```

Comment:
Can this functionality be removed if not needed anymore @Aprabhat19 @vspecky. We should not be making any database calls once the `call_connector_service` is entered, this feature of updating the mandate flow needs to be redesigned otherwise

Reply:
I assume this flow is not needed anymore,  and so can be removed. Now to update a mandate , we would have to update the `PaymentMethod`, which is being done in this [PR](https://github.com/juspay/hyperswitch/pull/4305)
CC: @vspecky @tinu-hareesswar

Reply:
I have removed the function